### PR TITLE
Of/cleanup solr4 ab code

### PR DESCRIPTION
**Pull request by oferrigni**

### CODE REVIEW

File: src/chef_wm_search.erl [hunk 1]
```diff
@@ -175,103 +170,20 @@ batch_size() ->
...
-                    ok;
-                {Solr1, Solr4} ->
-                    lager:log(debug, [{solr_diff, ?MODULE}], "~p",
-                              [?LOG_DIFF_TUPLE(Solr1, Solr4, Query, OrgName)])
-            end
+solr_search(Query) ->
+    try
+        chef_solr:search(Query)
+    catch
+        Error:Reason ->
```
Comment (sdelano) [inline src/chef_wm_search.erl:183]: Is this error handling new?

Comment (sdelano) [inline src/chef_wm_search.erl:183]: I don't see a spec for solr_query that returns the `{Error, Reason}` tuple.

Comment (sdelano) [inline src/chef_wm_search.erl:183]: `execute_solr_query` (the case for either solr1 or solr4) did not have this error handling code. I just want to make sure that we haven't introduced a bug by adding in error handling that we aren't catching here.

**Comment by sdelano** [comment]
Other than the remaining questions around error handling, :thumbsup:. Thanks!

File: src/chef_wm_search.erl (hunk 1 above)
Comment (oferrigni) [inline src/chef_wm_search.erl:183]: spawn_solr_query on https://github.com/opscode/chef_wm/pull/109/files#diff-fd88d5b9dfe15f5fb98e7061cf0a37faL274 had the error handling.  This error handling was introduced in response to the partial outage that occurred when solr1 was unavailable and searches were crashing the request process.

Comment (sdelano) [inline src/chef_wm_search.erl:183]: +1

**Pull request merged**